crates/metrics/src/metrics.rs

```diff
@@ -82,19 +95,84 @@ pub type DefaultMetricPublisher = crate::logger::LoggerMetricPublisher;
 /// be "$metric_prefix.latency".
 #[macro_export]
 macro_rules! with_latency_publishing {
-    ($metric_prefix:expr, $publisher:expr, $f:expr, $($args:expr),* ) => {{
+    ($metric_prefix:expr, $publisher:expr, $f:expr) => {{
         let clock = std::time::SystemTime::now();
 
-        let ret = ($f)($($args),*);
+        let ret = $f();
         let latency = clock.elapsed().unwrap().as_millis();
 
-        let metric_name = format!("{}.latency", $metric_prefix);
+        if latency == 0 {
+            ret
+        } else {
+            let metric_name = format!("{}.latency", $metric_prefix);
 
-        // TODO pass in stream id or not?
-        let metric = $crate::Metric::new(metric_name.as_str(), None,
+            // TODO pass in stream id or not?
+            let metric = $crate::Metric::new(metric_name.as_str(), None,
             Some(clock.into()), latency as f64);
-        $publisher.write().unwrap().publish(&metric);
-        ret
+            $publisher.write().unwrap().publish(&metric);
+            ret
+        }
+    }};
+    ($metric_prefix:expr, $publisher:expr, $f:expr, $($args:expr),+ ) => {{
+        let clock = std::time::SystemTime::now();
+
+        let ret = ($f)($($args),+);
+        let latency = clock.elapsed().unwrap().as_millis();
+
+        if latency == 0 {
+            ret
+        } else {
+            let metric_name = format!("{}.latency", $metric_prefix);
+
+            // TODO pass in stream id or not?
+            let metric = $crate::Metric::new(metric_name.as_str(), None,
+            Some(clock.into()), latency as f64);
+            $publisher.write().unwrap().publish(&metric);
+            ret
+        }
+    }}
+}
+
+/// Wraps a standard rust function with latency timing that is published to
+/// $publisher upon completion of self.f($args,*). The latency metric name will
+/// be "$metric_prefix.latency".
+#[macro_export]
+macro_rules! self_with_latency_publishing {
```

Comment:
@struktured it's a copy-n-paste but you said maybe this isn't necessary anymore anyway